src/Compilers/CSharp/Portable/Compilation/CSharpSemanticModel.cs

```diff
@@ -5282,6 +5282,56 @@ contextState switch
                 };
         }
 
+        protected override IParameterSymbol GetCorrespondingParameterCore(SyntaxNode node, CancellationToken cancellationToken)
+        {
+            if (node is ArgumentSyntax && node.Parent is ArgumentListSyntax list && node.Parent.Parent is ExpressionSyntax expression)
+            {
+                var argumentIndex = -1;
+                for (int i = 0; i < list.Arguments.Count; i++)
+                {
+                    if (list.Arguments[i] == node)
+                    {
+                        argumentIndex = i;
+                        break;
+                    }
+                }
+
+                if (argumentIndex == -1)
+                {
+                    return null;
+                }
+
+                var binder = GetEnclosingBinder(GetAdjustedNodePosition(node));
+                if (binder.BindExpression(expression, BindingDiagnosticBag.Discarded) is BoundCall boundCall)
```

Comment:
>if (binder.BindExpression(expression, BindingDiagnosticBag.Discarded) is BoundCall boundCall) [](start = 16, length = 93)

It feels like we are doing too much manual work here. Not using SemanticModel to get bound nodes, etc. Also, we might be "reinventing the wheel". IArgumentOperation already exposes corresponding parameter symbol. Perhaps, we don't need to expose this new API after all? Even if we decide that we do need it, I think that the logic should be shared between IArgumentOperation and this API. The behavior should be consistent across the breadth of scenarios, icluding error scenarios.

Reply:
> Perhaps, we don't need to expose this new API after all?

@AlekseyTs Not sure. I'd prefer if this could be discussed with the team before doing more work here.

Reply:
@333fred What do you think?

> Perhaps we don't need to expose this new API after all?

>> Not sure. I'd prefer if this could be discussed with the team before doing more work here.

---
In reply to: [599793432](https://github.com/dotnet/roslyn/pull/52054#discussion_r599793432) [](ancestors = 599793432)

Reply:
IMO, if customers want this, i would advise they use IOperation.  We can also show, through an example (or comment on an issue) how you could write GetCorrespondingParameter as an extension method that just defers to IOp to figure this out.

Reply:
@CyrusNajmabadi So should we close the PR and the linked issue?

Reply:
I would agree. `IArgumentOperation` is the way to go here, I don't think we need separate `SemanticModel` apis here.

Reply:
Should we close the PR and also #14003?